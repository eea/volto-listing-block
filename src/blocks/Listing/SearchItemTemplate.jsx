import cx from 'classnames';
import messages from '@eeacms/volto-listing-block/messages';
import PreviewImage from '@eeacms/volto-listing-block/PreviewImage';

const BodyText = ({ item, hasDescription }) => {
  return (
    <div className="listing-body">
      <h3 className={'listing-header'}>{item.title ? item.title : item.id}</h3>
      {hasDescription && (
        <p className={'listing-description'}>{item.description}</p>
      )}
    </div>
  );
};

const getStyles = (props) => {
  const { itemModel = {} } = props;
  const res = {};
  if (itemModel.maxDescription) {
    res[`max-${itemModel.maxDescription}-lines`] = true;
  }
  return res;
};

const BasicItem = (props) => {
  const { item, styles, className, itemModel = {} } = props;
  const { hasImage, hasDate, hasDescription } = itemModel;

  return (
    <div
      className={cx(
        'u-item listing-item result-item',
        styles?.theme,
        getStyles(props),
        className,
      )}
    >
      <div className="wrapper">
        <div className="slot-head">{item?.meta}</div>
        <div className="slot-top">
          <BodyText
            item={item}
            hasDescription={hasDescription}
            hasDate={hasDate}
          />
          {hasImage && (
            <PreviewImage
              item={item}
              style={{ marginLeft: 'auto' }}
              preview_image_url={item.preview_image_url}
            />
          )}
        </div>
        <div className="slot-bottom">{item?.extra}</div>
      </div>
    </div>
  );
};

export const SearchItemLayout = (props) => {
  return <BasicItem {...props} />;
};
SearchItemLayout.styleSchemaEnhancer = ({ schema, intl }) => {
  return schema;
};
