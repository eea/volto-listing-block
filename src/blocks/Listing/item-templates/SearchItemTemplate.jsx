import cx from 'classnames';
import PreviewImage from '@eeacms/volto-listing-block/PreviewImage';

const BodyText = ({ item, hasDescription }) => {
  return (
    <div className="listing-body">
      <h3 className={'listing-header'}>{item.title ? item.title : item.id}</h3>
      {hasDescription && (
        <p className={'listing-description'}>{item.description}</p>
      )}

      {item?.extra && <div className="slot-bottom">{item?.extra}</div>}
    </div>
  );
};

const getStyles = (props) => {
  const { itemModel = {} } = props;
  const res = {};
  if (itemModel.maxDescription) {
    res[`max-${itemModel.maxDescription}-lines`] = true;
  }
  if (itemModel.maxTitle) {
    res[`title-max-${itemModel.maxTitle}-lines`] = true;
  }
  return res;
};

const BasicItem = (props) => {
  const { item, styles, className, itemModel = {} } = props;
  const { hasImage, hasDate, hasDescription, imageOnRightSide } = itemModel;

  return (
    <div
      className={cx(
        'u-item listing-item result-item',
        styles?.theme,
        getStyles(props),
        className,
      )}
    >
      <div
        className={`wrapper ${imageOnRightSide ? 'right-image' : 'left-image'}`}
      >
        <div className="slot-head">{item?.meta}</div>
        <div className="slot-top">
          {hasImage ? (
            imageOnRightSide ? (
              <>
                <BodyText
                  item={item}
                  hasDescription={hasDescription}
                  hasDate={hasDate}
                />
                <div className="image-wrapper">
                  <PreviewImage
                    item={item}
                    preview_image_url={item.preview_image_url}
                  />
                </div>
              </>
            ) : (
              <>
                <div className="image-wrapper">
                  <PreviewImage
                    item={item}
                    preview_image_url={item.preview_image_url}
                  />
                </div>
                <BodyText
                  item={item}
                  hasDescription={hasDescription}
                  hasDate={hasDate}
                />
              </>
            )
          ) : (
            <BodyText
              item={item}
              hasDescription={hasDescription}
              hasDate={hasDate}
            />
          )}
        </div>
      </div>
    </div>
  );
};

export const SearchItemLayout = (props) => {
  return <BasicItem {...props} />;
};
