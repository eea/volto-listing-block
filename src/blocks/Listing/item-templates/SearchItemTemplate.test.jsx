import React from 'react';
import renderer from 'react-test-renderer';
import { SearchItemLayout } from './SearchItemTemplate';

describe('SearchItemTemplates', () => {
  it('renders correctly', () => {
    const item = {
      title: 'Search listing title',
      description: 'Search listing description',
      meta: 'Search listing meta',
      url: '/my-item-url',
    };

    const itemModel = {
      hasImage: true,
      hasDate: true,
      hasDescription: true,
      imageOnRightSide: false,
    };

    const component = renderer.create(
      <SearchItemLayout
        item={item}
        itemModel={itemModel}
        className="my-class"
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
