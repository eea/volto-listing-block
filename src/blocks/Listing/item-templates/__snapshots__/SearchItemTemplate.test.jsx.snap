// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItemTemplates renders correctly 1`] = `
<div
  className="u-item listing-item result-item my-class"
>
  <div
    className="wrapper left-image"
  >
    <div
      className="slot-head"
    >
      Search listing meta
    </div>
    <div
      className="slot-top"
    >
      <div
        className="image-wrapper"
      >
        <img
          alt="Search listing title"
          className="ui image"
          src="/my-item-url/@@images/image/preview"
        />
      </div>
      <div
        className="listing-body"
      >
        <h3
          className="listing-header"
        >
          Search listing title
        </h3>
        <p
          className="listing-description"
        >
          Search listing description
        </p>
      </div>
    </div>
  </div>
</div>
`;
